util/timeutil/time.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package timeutil
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// init initializes `locCache`.
+func init() {
+	LocCache = &locCache{}
+	LocCache.locMap = make(map[string]*time.Location)
+}
+
+// LocCache is a simple cache policy to improve the performance of 'time.LoadLocation'.
+var LocCache *locCache
+var localStr string
+var localOnce sync.Once
+var zoneSources = []string{
+	"/usr/share/zoneinfo/",
+	"/usr/share/lib/zoneinfo/",
+	"/usr/lib/locale/TZ/",
+}
+
+// locCache is a simple map with lock. It stores all used timezone during the lifetime of tidb instance.
+// Talked with Golang team about whether they can have some forms of cache policy available for programmer,
+// they suggests that only programmers knows which one is best for their use case.
+// For detail, please refer to: https://github.com/golang/go/issues/26106
+type locCache struct {
+	sync.RWMutex
+	// locMap stores locations used in past and can be retrieved by a timezone's name.
+	locMap map[string]*time.Location
+}
+
+func initLocalStr() {
+	// consult $TZ to find the time zone to use.
+	// no $TZ means use the system default /etc/localtime.
+	// $TZ="" means use UTC.
+	// $TZ="foo" means use /usr/share/zoneinfo/foo.
+	tz, ok := syscall.Getenv("TZ")
+	if ok && tz != "" {
+		for _, source := range zoneSources {
+			if _, err := os.Stat(source + tz); os.IsExist(err) {
+				localStr = tz
+				break
+			}
+		}
+		// } else if tz == "" {
+		// localStr = "UTC"
+	} else {
+		path, err := filepath.EvalSymlinks("/etc/localtime")
+		if err == nil {
+			localStr, err = getTZNameFromFileName(path)
+			if err == nil {
+				return
+			}
+			log.Errorln(err)
+		}
+		log.Errorln(err)
+	}
+	localStr = "System"
```

Comment:
Need to fallback to check "/etc/timezone", where system stores timezone name in Debian distribution and its derives, which includes Alpine Linux, the distribution used in our docker image.

Reply:
https://github.com/pingcap/tidb/pull/7638/files#diff-512da1d8fff778df5097186a394aa101R71 Yes. See this line.

Reply:
No, what I refers to is "/etc/timezone", not "/etc/localtime".

Reply:
Oh, my bad. I overlooked. I need dig into `/etc/timezone`